Automattic/newspack-custom-content-migrator
# Clean up Update Recipe

## Changes made in this PR

Cleans up and simplifies the Update Recipe structure and output:
- [renames historic VaultPress wording to Jetpack](https://github.com/Automattic/newspack-custom-content-migrator/commit/8ee75a4a847557e0db7c1c8393067d301a19a5bd)
- [removes unnecessary if-s](https://github.com/Automattic/newspack-custom-content-migrator/commit/0e0153d3232c68902c1e2fdfc6e53618ff33b51e)
- [removes one-liner functions](https://github.com/Automattic/newspack-custom-content-migrator/commit/e5c530998a94a0716ef52378db1674ea81dedb5e)
- [simplifies config variables](https://github.com/Automattic/newspack-custom-content-migrator/commit/500a24c26279accf21b2113fa78bf28af254bdce)
- [`search-replace` is now used if found locally, and downloaded only if not](https://github.com/Automattic/newspack-custom-content-migrator/commit/8ac4ffd702c871f4f4acd6fd5b5161d375d8790d)
- [drops `live_` and `staging_` temp DB tables when done](https://github.com/Automattic/newspack-custom-content-migrator/commit/9dde9aa6107adee941d8326d007bba41cc03a0ff)
- [improves wording in many output messages](https://github.com/Automattic/newspack-custom-content-migrator/commit/9b592ac88c01b701366a167cb13e2a866cd408e4)

## Changes this brings to the Launch content migration

No significant changes in logic were introduced, nor any significant changes in how this script is used. This script is used the exact same way we used it until to now.

## Steps to test
- to emulate the launch content migration, create a test "Launch site" by cloning one of our smallest live sites onto a test domain (similarly as we clone the Staging site to create the Launch site). Call it something in the lines of `nccm-test-launch.newspackstaging.com`
- install the PasswordProtected on this test Launch site (to protect it from unauthorized access or wrong SE indexing)
- create the `/tmp/test_launch` dir
- get the actual Jetpack Rewind archive link from the actual live site you used for cloning base, and download it to the `/tmp/test_launch` dir
- pull this branch, run `composer update` and install this plugin on the test Launch site
- edit the update recipe script, and set the variables at the top as usual:
    - `TABLE_PREFIX`
    - `DB_DEFAULT_CHARSET`
    - `LIVE_JETPACK_ARCHIVE`
    - `LIVE_SQL_DUMP_HOSTNAME_REPLACEMENTS` -- same as ever, for the purposes of this test, key is the live domain and value is this test Launch site's hostname
    - `STAGING_SITE_HOSTNAME` -- this is a new variable, and as the comment above it says it's `site from which this site was cloned. In our normal Launch day content migration process, this would be the hostname of the Staging site, but since to simplify this test we've modified creation of the sites and cloned the test Launch site directly from the live site, just use the live site's domain
    - `TEMP_DIR` can either be left as it is, or better yet changed to `/tmp/test_launch/temp`
- run the update recipe script, confirm all the output was correct, and no unexpected warnings or errors occurred
- delete the test Launch site
